lib/OpenQA/WebAPI/Auth/OAuth2.pm

```diff
@@ -24,39 +24,69 @@ sub auth_setup {
     my ($self) = @_;
     $self->config($self->app->config->{oauth2});
     croak 'No OAuth2 provider selected' unless my $provider = $self->config->{provider};
-    croak "Provider $provider not supported" unless $provider eq 'github';
+    my $prov_args = {
+        key    => $self->config->{key},
+        secret => $self->config->{secret},
+    };
+    # I'm afraid I don't quite get where I should tuck this away so that I get
+    # to use it in auth_login, so I'm doing this FIXME_oauth2_ nastiness for
+    # now. I had hoped to tack this onto $prov_args somehow, but I don't know
+    # how to then access that later.
+    if ($provider eq 'github') {
+        $self->config->{FIXME_oauth2_user_url} = 'https://api.github.com/user';
```

Comment:
My instincts would tell me to put this into a yaml file or a perl hash to avoid all of the assignments, and simply declare the respective values under the provider name and look it up once.